pagopa/io-app
# chore: [IOBP-138] Replace legacy bottom sheets

## Short description
This PR removes the use of `useIOLegacyBottomSheet` from wallet and payment sections of the app.
This PR also includes a fix for `useIOBottomSheet` bottom padding in case it has a footer component.

## List of changes proposed in this pull request
- Replaced `useIOLegacyBottomSheet` in wallet and payments sections
- Removed `marginTop` from `useIOBottomSheetAutoresizableModal ` and replaced with a dynamic snap point to avoid the modal to go behind the top notch, if any.
- Added `fullScreen` parameter to `useIOBottomSheetAutoresizableModal` to bypass the dynamic snap point

## How to test
- Tests should solve correctly.
- To test the bottom sheets, go to **Profile → Design System → Bottom Sheet**.
- To test the replaced bottom sheets you should:
  - **Wallet**: 
    - `WALLET_HOME`: click on "+" icon
    - `WALLET_CREDIT_CARD_DETAIL`: click on remove payment method
    - `WALLET_ONBOARDING_PAYPAL_SEARCH_PSP`: click on link in the description and info icon on any item
    - `PAYMENT_TRANSACTION_SUMMARY`: click on info icon near amount
  - **CGN**:  
    - `CGN_DETAILS`: click on info icon near EYCA number
    - `CGN_MERCHANT_DETAIL`: click on any item in the list
  - **IDPay**:
    - `IDPAY_ONBOARDING_PDND_ACCEPTANCE`: click on info icon 
    - `IDPAY_DETAILS_BENEFICIARY`: click on "Show more"
    - `IDPAY_UNSUBSCRIPTION_CONFIRMATION`: click on "Continue"
    - `IDPAY_CONFIGURATION_INSTRUMENTS`: add an instrument
